### PR TITLE
Tidying up and improving the React logic for Stripe components

### LINKED ISSUE #9409: Tweaks to session refresh logic
### Description

If a Stripe AccountSession runs for too long, it may expire. We need to ensure our logic can handle this for embedded KYC.

### Acceptance criteria

* Ensure our account session can handle needing to refresh if it expires: https://docs.stripe.com/connect/get-started-connect-embedded-components#refresh-the-client-secret

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #9409 

#### Changes proposed in this Pull Request

* Moving API calls to the `utils` file to simplify the logic
* Removing some unneeded code and tidying up the error handling.
* Some updates to the type definitions for consistency
* Some updates to how we handle sessions

#### Testing instructions

* Remove any existing account you have on your store (or using a JN site).
* In the dev tools, ensure `Embedded KYC` is enabled in the feature selection.
* Open the network tab on your dev tools in your browser.
* Start an onboarding. When you get to the Embedded KYC step, look for the request to the `/onboarding/kyc/session` endpoint. Check the value of `expiresAt` and make a note of it.
* Wait until _after_ the `expiresAt` expires before proceeding with the embedded KYC onboarding.
* You should not see an error about the session being expired. The onboarding should work as expected.
* Also try leaving the onboarding midway through embedded KYC and going back to it (via the Connect page) - should work as expected.
* After the onboarding, you should be redirected to the Overview page.
* Do another onboarding (from the start), this time with a Progressive Account and ensure that the progressive onboarding also works as expected. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9425` or branch name `dev/session-fixes` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/session-fixes) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 603b0933f46ba35c4b867bbb2ed09e1995b5c60b
- Build time: 2024-09-12 13:58:27 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/onboarding/utils.ts [hunk 1]
```diff
@@ -42,6 +55,86 @@ export const getBusinessTypes = (): Country[] => {
...
+		method: 'GET',
+	} );
+};
+
+/**
+ * Make an API request to finalize the onboarding process.
+ *
+ * @param urlSource The source URL.
+ */
+export const finalizeOnboarding = async ( urlSource: string ) => {
```
Comment (oaratovskyi) [inline client/onboarding/utils.ts:85]: we could type-hint `Promise< FinalizeOnboardingResponse >` here

File: client/onboarding/types.ts [hunk 2]
```diff
@@ -65,6 +56,10 @@ export interface MccsDisplayTreeItem {
 	keywords?: string[];
 }
 
+export interface AccountKycData {
```
Comment (oaratovskyi) [inline client/onboarding/types.ts]: it seems to be not used, maybe remove it? 🙂

File: client/onboarding/steps/embedded-kyc.tsx [hunk 3]
```diff
@@ -1,51 +1,40 @@
 /**
  * External dependencies
  */
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
```
Comment (oaratovskyi) [inline client/onboarding/steps/embedded-kyc.tsx]:
```suggestion
import React, { useCallback, useEffect, useState } from 'react';
```
```suggestion
import React, { useCallback, useEffect, useRef, useState } from 'react';
```

Comment (oaratovskyi) [reply client/onboarding/steps/embedded-kyc.tsx]: remove as not used

**Review by oaratovskyi — APPROVED** [review]
The changes look amazing, nice refactoring! Works well :shipit:

**Comment by dmallory42** [comment]
Thanks for the review @oaratovskyi, I addressed your comments!

**Pull request merged**